### PR TITLE
feat: Support for Microsoft Playwright

### PR DESCRIPTION
**Pull request by phileba**
Instead of running the test on Chrome only, now we should support users running a test on other kinds of browsers like Firefox, Safari. So Playwright is a good choice for replacing.
https://github.com/microsoft/playwright

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
 		"ksuid": "^1.2.0",
 		"micromatch": "^4.0.2",
 		"puppeteer": "^2.1.1",
```
Comment (ai-flood) [inline packages/core/package.json:50]: Why haven't we removed puppeteer yet?

File: packages/core/src/page/types.ts [hunk 2]
```diff
@@ -19,16 +13,23 @@ export interface LocatorBuilder {
 	toString(): string
 }
 
+export type EvaluateFn<T = any> = string | ((arg1: T, ...args: any[]) => any)
+export interface PageGoToOptions {
```
Comment (HieuLeKMS) [inline packages/core/src/page/types.ts:17]: I think `NavigationOptions` sounds better, and that's what we're currently using, with puppeteer

File: examples/internet-herokuapp/test.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+class Test {}
```
Comment (HieuLeKMS) [inline examples/internet-herokuapp/test.ts:1]: what's this for?

File: packages/core/src/runtime/Settings.ts [hunk 4]
```diff
@@ -48,7 +47,7 @@ export type ConsoleMethod = 'log' | 'info' | 'debug' | 'warn' | 'error'
  * puppeteer | (Default) The browser bundled with [puppeteer]. It is a curated version of [chromium](https://www.chromium.org) (the open source version of Google Chrome). Using the puppeteer-bundled Chromium ensures the best compatibility with puppeteer, but lacks some features such as video support.
  * stable | The latest version of [Google Chrome](https://www.chromium.org/). Google Chrome has more features than chromium, but isn't tested as thoroughly against puppeteer, which can result in intermittent errors. If you don't need the extra features, please use `bundled`.
  */
-export type ChromeVersion = 'puppeteer' | 'stable'
+export type BrowserType = BROWSER_TYPE
```
Comment (HieuLeKMS) [inline packages/core/src/runtime/Settings.ts:50]: The description above should be updated

Comment (HieuLeKMS) [reply packages/core/src/runtime/Settings.ts:50]: Also I don't see BrowserType is used anywhere

Comment (phileba) [reply packages/core/src/runtime/Settings.ts:49]: you can use `BrowserType` in TestSetting, to define which browser will be run for the test

File: examples/internet-herokuapp/test.ts (hunk 3 above)
Comment (phileba) [reply examples/internet-herokuapp/test.ts:1]: it's just a simple test file for the `18-File_Upload.ts` => `Test: 03 - Choose File`

**Pull request merged**